integrations/openaiwrapper.py

```diff
@@ -20,53 +22,67 @@ def __init__(self, api_key, timeout=10):
         self.api_key = api_key
         openai.api_key = api_key
         self.timeout = timeout
-        self.cache = {}
 
+    @memoize_to_sqlite(func_name="get_embedding", filename="openai_embedding_cache.db")
     def get_embedding(self, text):
         """
         Retrieves the embedding for the given text.
 
         :param text: The text for which embedding is required.
         :return: The embedding for the given text.
         """
-        if text in self.cache:
-            return self.cache[text]
-
         start_time = time.time()
         retries = 0
 
         while time.time() - start_time < self.timeout:
             try:
-                embedding = openai.Embedding.create(input=text, engine="text-embedding-ada-002")
-                self.cache[text] = embedding
-                return embedding
+                return openai.Embedding.create(input=text, engine="text-embedding-ada-002")
             except openai.error.OpenAIError as e:
                 logging.error(f"OpenAI API error: {e}")
                 retries += 1
                 if retries >= MAX_RETRIES:
                     raise
                 time.sleep(RETRY_SLEEP_DURATION)
-        
+                #for those using the free version of openai
+                #if f"{e}".startswith("Rate limit"):
+                #   print("Rate limit reached...  sleeping for 20 seconds")
+                #   start_time+=20
+                #   time.sleep(20)
         raise TimeoutError("API call timed out")
 
+    @memoize_to_sqlite(func_name="chat_completion", filename="openai_chat_cache.db")
```

Comment:
Yes, I was not sure if we should cache the results for the chat_completion part, but thought I use it for testing and debugging.  I also was not sure exactly what was going on, so caching the results allowed me to learn what was going on without accessing openai for the same calls, etc.  
I will remove it, and it will be easy to add back in for testing, etc, if we want.